src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormat.cs

```diff
@@ -681,8 +681,16 @@ private static bool IsUseGenitiveForm(ReadOnlySpan<char> format, int index, int
                             }
                             else
                             {
-                                string fmtPattern = "D" + tokenLen.ToString();
-                                result.Append(year.ToString(fmtPattern, CultureInfo.InvariantCulture));
+                                string fmtPattern;
+                                switch (tokenLen)
+                                {
+                                    // "yyy", "yyyy", and "yyyyy" are explicitly highlighted in docs.
+                                    case 3: fmtPattern = "D3"; break;
+                                    case 4: fmtPattern = "D4"; break;
+                                    case 5: fmtPattern = "D5"; break;
+                                    default: fmtPattern = "D" + tokenLen.ToString(); break;
+                                }
```

Comment:
I think this can work if called FormatDigits(result, year, tokenLen, true)

note the last true parameter value. 

but we'll need to do some testing with different token lengths to ensure we are not missing anything. 

note that FormatDigits is limited to 16 digits so it will be good to test with y token length 15 or so :-) I know this is extremely unlikely scenario but we should ensure we are not breaking any scenario. 


---
In reply to: [249314574](https://github.com/dotnet/coreclr/pull/22111#discussion_r249314574) [](ancestors = 249314574)